From 68b310d318c6a470021655c0d48b30a13c0797ac Mon Sep 17 00:00:00 2001
From: Tomaz Zaman <tomaz@mono.si>
Date: Sun, 19 Apr 2026 15:24:51 +0200
Subject: [PATCH 6/7] arm: dts: add mono-gateway-dk.dts

Adds the U-Boot device tree for the Mono Gateway Development Kit. It
references the LS1046A SoC dtsi from upstream and adds the board's
peripherals: I2C buses with their muxes and downstream chips, the
SerDes/PCIe lanes, and the SD/eMMC controllers. The dts file is
wired into the build by the next patch.

Upstream-Status: Inappropriate [product-specific]

Signed-off-by: Tomaz Zaman <tomaz@mono.si>
---
 arch/arm/dts/mono-gateway-dk.dts | 176 +++++++++++++++++++++++++++++++
 1 file changed, 176 insertions(+)
 create mode 100644 arch/arm/dts/mono-gateway-dk.dts

diff --git a/arch/arm/dts/mono-gateway-dk.dts b/arch/arm/dts/mono-gateway-dk.dts
new file mode 100644
index 00000000000..05c2381f244
--- /dev/null
+++ b/arch/arm/dts/mono-gateway-dk.dts
@@ -0,0 +1,176 @@
+// SPDX-License-Identifier: GPL-2.0+ OR X11
+/*
+ * Device Tree Include file for Gateway Development Kit.
+ *
+ * Copyright 2025 Mono Technologies Inc.
+ *
+ */
+
+#include <dt-bindings/leds/common.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/dts-v1/;
+#include "fsl-ls1046a.dtsi"
+
+/ {
+	model = "Mono Gateway Development Kit";
+
+	aliases {
+		spi0 = &qspi;
+		serial0 = &duart0;
+		serial1 = &duart1;
+		mdio0 = &mdio;
+        i2c0 = &i2c0;
+        i2c1 = &i2c1;
+        i2c2 = &i2c2;
+        i2c3 = &i2c3;  /* Force i2c@21b0000 to always be bus 3 */
+    };
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+};
+
+&duart0 {
+	bootph-all;
+	status = "okay";
+};
+
+&duart1 {
+	bootph-all;
+	status = "okay";
+};
+
+&i2c0 {
+	status = "okay";
+};
+
+&i2c1 {
+	status = "okay";
+};
+
+&i2c2 {
+    status = "okay";
+    
+    i2c_mux: i2c-mux@70 {
+        compatible = "nxp,pca9546";
+        #address-cells = <1>;
+        #size-cells = <0>;
+        reg = <0x70>;
+        
+        i2c@2 {  /* Channel 2 */
+            #address-cells = <1>;
+            #size-cells = <0>;
+            reg = <2>;
+            
+            pcf2131: rtc@53 {
+                compatible = "nxp,pcf2131";
+                reg = <0x53>;
+            };
+        };
+    };
+};
+
+&i2c3 {
+    status = "okay";
+    
+    eeprom@50 {
+        compatible = "atmel,24c32";   /* AT24CS32 - 32Kbit (4KB) */
+        reg = <0x50>;
+        pagesize = <32>;              /* 32-byte page writes */
+        size = <4096>;                /* 4KB (32Kbit) total size */
+    };
+};
+
+&usb1 {
+	status = "disabled";
+};
+
+&usb2 {
+	status = "disabled";
+};
+
+&qspi {
+	status = "okay";
+
+	mt25qu512abb: flash@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "jedec,spi-nor";
+		spi-max-frequency = <50000000>;
+		reg = <0>;
+		spi-tx-bus-width = <1>;
+        spi-rx-bus-width = <4>;
+	};
+
+};
+
+&esdhc {
+	mmc-hs200-1_8v;
+	bus-width = <4>;
+	status = "okay";
+};
+
+#include "fsl-ls1046-post.dtsi"
+
+&fman0 {
+	ethernet@e0000 {
+		status = "disabled";
+	};
+
+	ethernet@e4000 {
+		status = "disabled";
+	};
+
+	ethernet@e6000 {
+		status = "disabled";
+	};
+
+	ethernet@e8000 {
+		phy-handle = <&sgmii_phy0>;
+		phy-connection-type = "sgmii";
+		status = "okay";
+	};
+
+	ethernet@ea000 {
+		phy-handle = <&sgmii_phy1>;
+		phy-connection-type = "sgmii";
+		status = "okay";
+	};
+
+	ethernet@e2000 {
+		phy-handle = <&sgmii_phy2>;
+		phy-connection-type = "sgmii";
+		status = "okay";
+	};
+
+	ethernet@f0000 {
+		fixed-link = <0 1 10000 0 0>;
+		phy-connection-type = "xgmii";
+		status = "okay";
+	};
+
+	ethernet@f2000 {
+		fixed-link = <0 1 10000 0 0>;
+		phy-connection-type = "xgmii";
+		status = "okay";
+	};
+
+	mdio@fc000 {
+		status = "disabled";
+	};
+
+	mdio: mdio@fd000 {
+		sgmii_phy0: ethernet-phy@0 {
+			reg = <0x0>;
+		};
+
+		sgmii_phy1: ethernet-phy@1 {
+			reg = <0x1>;
+		};
+
+		sgmii_phy2: ethernet-phy@2 {
+			reg = <0x2>;
+		};
+	};
+};
-- 
2.47.3

